ui-v2/app/services/dom.js

```diff
@@ -2,25 +2,36 @@ import Service from '@ember/service';
 import { getOwner } from '@ember/application';
 import { get } from '@ember/object';
 
+// selecting
 import qsaFactory from 'consul-ui/utils/dom/qsa-factory';
-// TODO: Move to utils/dom
-import getComponentFactory from 'consul-ui/utils/get-component-factory';
+// TODO: sibling and closest seem to have 'PHP-like' guess the order arguments
+// ie. one `string, element` and the other has `element, string`
+// see if its possible to standardize
+import sibling from 'consul-ui/utils/dom/sibling';
+import closest from 'consul-ui/utils/dom/closest';
+import getComponentFactory from 'consul-ui/utils/dom/get-component-factory';
+
+// events
 import normalizeEvent from 'consul-ui/utils/dom/normalize-event';
 import createListeners from 'consul-ui/utils/dom/create-listeners';
+import clickFirstAnchorFactory from 'consul-ui/utils/dom/click-first-anchor';
 
 // ember-eslint doesn't like you using a single $ so use double
 // use $_ for components
 const $$ = qsaFactory();
 let $_;
+const clickFirstAnchor = clickFirstAnchorFactory(closest);
 export default Service.extend({
   doc: document,
   init: function() {
```

Comment:
Along the same idea as `viewport` above I'm thinking of exposing `document` as just `dom.document()`, I'm not referring to `document` that much in my codebase but I may aswell go the whole hog here?